watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -40,58 +42,80 @@
     PressureChangeType,
 )
 from watertap.unit_models.pressure_exchanger import PressureExchanger
-from watertap.unit_models.pressure_changer import Pump
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
 from watertap.core.util.initialization import assert_degrees_of_freedom
 from watertap.costing import WaterTAPCosting
 
 
-def main():
+def main(erd_type="pressure_exchanger"):
     # set up solver
     solver = get_solver()
 
     # build, set, and initialize
-    m = build()
-    set_operating_conditions(m, water_recovery=0.5, over_pressure=0.3, solver=solver)
+    m = build(erd_type=erd_type)
+    set_operating_conditions(m)
     initialize_system(m, solver=solver)
 
-    # simulate and display
+    # optimize and display
+    optimize_set_up(m)
     solve(m, solver=solver)
+
     print("\n***---Simulation results---***")
     display_system(m)
     display_design(m)
-    display_state(m)
+    if erd_type == "pressure_exchanger":
+        display_state(m)
+    else:
+        pass
 
-    # optimize and display
-    optimize_set_up(m)
-    optimize(m, solver=solver)
-    print("\n***---Optimization results---***")
-    display_system(m)
-    display_design(m)
-    display_state(m)
+    return m
 
 
-def build():
+def build(erd_type="pressure_exchanger"):
     # flowsheet set up
     m = ConcreteModel()
+    m.erd_type = erd_type
     m.fs = FlowsheetBlock(default={"dynamic": False})
     m.fs.properties = props.NaClParameterBlock()
     m.fs.costing = WaterTAPCosting()
 
     # unit models
     m.fs.feed = Feed(default={"property_package": m.fs.properties})
-    m.fs.S1 = Separator(
-        default={"property_package": m.fs.properties, "outlet_list": ["P1", "PXR"]}
-    )
     m.fs.P1 = Pump(default={"property_package": m.fs.properties})
-    m.fs.PXR = PressureExchanger(default={"property_package": m.fs.properties})
-    m.fs.P2 = Pump(default={"property_package": m.fs.properties})
-    m.fs.M1 = Mixer(
-        default={
-            "property_package": m.fs.properties,
-            "momentum_mixing_type": MomentumMixingType.equality,  # booster pump will match pressure
-            "inlet_list": ["P1", "P2"],
-        }
-    )
+
+    if erd_type == "pressure_exchanger":
```

Comment:
Might be overkill, but maybe we could implement as an enumerated class

Reply:
Specifically, a StrEnum for ease: `idaes.core.util.misc.StrEnum`. Probably best practice on flowsheets, though we obviously don't do this everywhere else.

Reply:
Now implemented the configuration choice using StrEnum.